tests/testSimpleTable.py

```diff
@@ -434,23 +434,23 @@ def testTicket2850(self):
     def testTicket2894(self):
         """Test boolean-array indexing of catalogs"""
         schema = lsst.afw.table.Schema()
-        key = schema.addField(lsst.afw.table.Field[int]("i", "doc for i"))
+        key = schema.addField(lsst.afw.table.Field[np.int32]("i", "doc for i"))
         cat1 = lsst.afw.table.BaseCatalog(schema)
         cat1.addNew().set(key, 1)
         cat1.addNew().set(key, 2)
         cat1.addNew().set(key, 3)
-        cat2 = cat1[numpy.array([True, False, False], dtype=bool)]
-        self.assertFloatsEqual(cat2[key], numpy.array([1], dtype=int))
+        cat2 = cat1[np.array([True, False, False], dtype=bool)]
+        self.assertFloatsEqual(cat2[key], np.array([1], dtype=np.int32))
         self.assertEqual(cat2[0], cat1[0])  # records compare using pointer equality
-        cat3 = cat1[numpy.array([True, True, False], dtype=bool)]
-        self.assertFloatsEqual(cat3[key], numpy.array([1, 2], dtype=int))
-        cat4 = cat1[numpy.array([True, False, True], dtype=bool)]
-        self.assertFloatsEqual(cat4.copy(deep=True)[key], numpy.array([1, 3], dtype=int))
+        cat3 = cat1[np.array([True, True, False], dtype=bool)]
+        self.assertFloatsEqual(cat3[key], np.array([1, 2], dtype=np.int32))
+        cat4 = cat1[np.array([True, False, True], dtype=bool)]
+        self.assertFloatsEqual(cat4.copy(deep=True)[key], np.array([1, 3], dtype=np.int32))
 
     def testTicket2938(self):
         """Test heterogenous catalogs that have records from multiple tables"""
         schema = lsst.afw.table.Schema()
-        schema.addField("i", type=int, doc="doc for i")
+        schema.addField("i", type=np.int32, doc="doc for i")
```

Comment:
Why the change from bare `int` to `np.int32` (and similarly for `float`)? Do we expect users to always use the numpy types now? I hope not, as it will be a lot of retraining.

Also, I would expect `int` to be equivalent to `np.int64` but based on this and other similar changes it is not. Quite surprising.